### DISCUSSION TITLE
`.discriminators` property for nested-discriminators / discriminators on subdocument paths

### DISCUSSION THREAD

**Discussion by hasezoey** [Automattic/mongoose#12159] [Q&A] [answered]
Is there currently a property like `Model.prototype.discriminators` to get all the discriminator models on a subdocument / array subdocument path where the list is actually models instead of a schema (like `Model.prototype.discriminators` and not `Schema.prototype.discriminators`)?

**Comment by vkarpov15** [comment, marked as answer]
Not public currently. Document array schematypes have `casterConstructor.discriminators`, and single nested subdocument schematypes have `caster.discriminators`, that would be how to get the current discriminators defined on a document array or subdocument.

What are you trying to build?

**Reply by hasezoey** [reply]
> Document array schematypes have casterConstructor.discriminators, and single nested subdocument schematypes have caster.discriminators, that would be how to get the current discriminators defined on a document array or subdocument.

wouldnt those just return the schemas for the discriminators?

> What are you trying to build?

currently in typegoose i have a way to specify nested-discriminators but had not a way to get the resulting models, and before i implement it on the typegoose side i wanted to ask if there is something already in place to get the already created nested-discriminator models again, like a `Model.prototype.discriminators`

**Reply by vkarpov15** [reply]
`casterContructor.discriminators` contains the constructor that Mongoose uses to create subdocuments, so that's the "model" equivalent for subdocument discriminators. Is that helpful?

**Reply by hasezoey** [reply]
thanks for answering, i dont quite know anymore what i found back then when creating this discussion, but looking over it now again (in code), it is how i would expect it to be (and how you had explained it)
